Create what-just-happened_test_plan.md

noaOrMlnx opened the pull request:
Create what-just-happened test plan file.

noaOrMlnx committed:
Create what-just-happened_test_plan.md

stepanblyschak commented on docs/what-just-happened/what-just-happened_test_plan.md:
````diff
@@ -0,0 +1,145 @@
+
+For this purpose, a pre test fixture will do the following:
+
+- Execute ```show feature``` command on DUT. 
````
I would suggest to implement "feature_facts" as ansible module which can be used by ansible and pytest.

stepanblyschak commented on docs/what-just-happened/what-just-happened_test_plan.md:
````diff
@@ -0,0 +1,145 @@
+what-just-happened  enabled
+```
+
+- If one of the checks on previous phase is indicated what-just-happened as not available, skip the test. 
````
You probably need to verify that debug mode is enabled by looking at "WJH" table and verify or set channels in WJH_CHANNEL. For forwarding - you need to ensure that the relevant channel exists in DB.

stepanblyschak commented on docs/what-just-happened/what-just-happened_test_plan.md:
```diff
@@ -0,0 +1,145 @@
+## Test plan
+
+### Overview
+The test will use test cases appear in a file named drop_packets.py.
```
Can you emphasise the relatio between this new test and existing drop counters test? Will they share the same set of test cases?

stepanblyschak commented on docs/what-just-happened/what-just-happened_test_plan.md:
```diff
@@ -0,0 +1,145 @@
+
+After each test case generates the packet, it calls do_test() function which implemented inside test_what_just_happened.py file. 
+
+do_test() function will send the packet, check if packet indeed dropped, and parse the output of 'show what-just-happened' command. 
```
Does the test verifies that there is only one signle entry in the table? E.g. does running background traffic affects the test result?

noaOrMlnx committed:
Fix review comments

- Add an implementation of ansible module/ features for pre test checks.
- Add a description of the relation with drop_counters test.
- Add clarification NOTEs.

noaOrMlnx committed:
Update what-just-happened_test_plan.md

noaOrMlnx committed:
Update what-just-happened_test_plan.md

- Add tunnel test case.
- Add support in aggregated packets.
